packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx

```diff
@@ -290,131 +299,133 @@ export function RedPacketERC721Form(props: RedPacketERC721FormProps) {
-                </div>
-                {contract && balance && !loadingOwnerList ? (
-                    <>
-                        <Typography className={classes.unapprovedTip}>{t.nft_unapproved_tip()}</Typography>
-                        <Typography className={classes.approveAllTip}>{t.nft_approve_all_tip()}</Typography>
-                    </>
-                ) : null}
-            </Box>
-            <Box style={{ position: 'absolute', bottom: 0, width: '100%' }}>
-                <PluginWalletStatusBar>
-                    <ChainBoundary expectedPluginID={NetworkPluginID.PLUGIN_EVM} expectedChainId={chainId}>
-                        <WalletConnectedBoundary>
-                            <EthereumERC721TokenApprovedBoundary
-                                validationMessage={validationMessage}
-                                owner={account}
-                                contractDetailed={contract}
-                                classes={{ approveButton: classes.approveButton }}
-                                operator={RED_PACKET_NFT_ADDRESS}>
-                                <ActionButton
-                                    style={{ height: 40, padding: 0, margin: 0 }}
-                                    size="large"
-                                    disabled={!!validationMessage}
-                                    fullWidth
-                                    onClick={() => setOpenConfirmDialog(true)}>
-                                    {t.next()}
-                                </ActionButton>
-                            </EthereumERC721TokenApprovedBoundary>
-                        </WalletConnectedBoundary>
-                    </ChainBoundary>
-                </PluginWalletStatusBar>
-            </Box>
-            {open ? (
+            {openSelectNFTDialog ? (
                 <SelectNftTokenDialog
-                    open={open}
-                    onClose={() => setOpen(false)}
+                    onClose={() => setOpenSelectNFTDialog(false)}
                     contract={contract}
                     existTokenDetailedList={tokenDetailedList}
                     setExistTokenDetailedList={setExistTokenDetailedList}
                     tokenDetailedOwnerList={tokenDetailedOwnerList}
                     loadingOwnerList={loadingOwnerList}
                 />
-            ) : null}
-            {openConfirmDialog && contract ? (
+            ) : openNFTConfirmDialog && contract ? (
                 <RedpacketNftConfirmDialog
                     message={message}
                     contract={contract}
-                    open={openConfirmDialog}
                     tokenList={tokenDetailedList}
-                    onBack={() => setOpenConfirmDialog(false)}
+                    onBack={() => setOpenNFTConfirmDialog(false)}
                     onClose={onClose}
                 />
-            ) : null}
+            ) : (
+                <>
```

Comment:
Does it better to have a sub-component for this?

Reply:
Refactored as `if (...) return ...` way.